docs/source/typing_anti_pitch.rst

```diff
@@ -0,0 +1,80 @@
+.. _typing-anti-pitch:
+
+Reasons to avoid static type checking
+=====================================
+
+In the words of :pep:`484`::
+
+    It should also be emphasized that Python will remain a dynamically typed language, and the
+    authors have no desire to ever make type hints mandatory, even by convention.
+
+The idea that dynamism in Python is a strength of the language is reflected in the fact that
+Python's type system is gradual. See :pep:`483` for details, but the long and short of this is
+that you can add static types to your codebase only to the extent that you want to, and static
+type checkers are designed to be able to put up with this.
+
+It's also worth noting that "static type checking" encompasses a spectrum of possible degrees of
+strictness. On the one hand, you can set yourself up so that your type checker does almost nothing.
+On the other -- well, I love type checking, but I would quit Python if I had to enable all
+possible strictness checks that type checkers offer.
+
+Anway, with all that said, here's a list of possible reasons to not use static type checking
```

Comment:
Could you add one point on readability/complexity? Readability is subjective but is one common reason that good type hints for a function may make signature too long/harder to understand.

Reply:
Yeah. Readability can often be mitigated using type aliases, but that doesn't address the complexity issue.

Reply:
Added a commit for this: https://github.com/python/typing/pull/1477/commits/b2a059846aca1275ed9565f2006792590c10e7bc